base/deprecated.jl

```diff
@@ -440,3 +440,11 @@ export float32_isvalid, float64_isvalid
 @deprecate (&)(x::Char, y::Char)  Char(UInt32(x) & UInt32(y))
 @deprecate (|)(x::Char, y::Char)  Char(UInt32(x) | UInt32(y))
 @deprecate ($)(x::Char, y::Char)  Char(UInt32(x) $ UInt32(y))
+
+# 11241
+
+@deprecate is_valid_char(ch::Union(Unsigned, Integer, Char)) isvalid(Char, ch)
+@deprecate is_valid_ascii(str::Union(Vector{UInt8}, ByteString)) isvalid(ASCIIString, str)
```

Comment:
It should be possible to split
```julia
@deprecate is_valid_ascii(str::Union(Vector{UInt8}, ByteString)) isvalid(ASCIIString, str)
```
into
```julia
@deprecate is_valid_ascii(str::Union(Vector{UInt8}, UTF8String)) isvalid(ASCIIString, str)
@deprecate is_valid_ascii(str::ASCIIString)) isvalid(str)
```
since `ByteString` is already a union.